napari/layers/points/points.py

```diff
@@ -2043,7 +2043,7 @@ def _set_highlight(self, force: bool = False) -> None:
             pos = None
 
         self._highlight_box = pos
-        if prev_stored != self._selected_data_stored:
+        if prev_stored != self._selected_data_stored or self._is_selecting:
```

Comment:
```suggestion
        if not force:
```

Or save 
```
data_changed = (
            self.selected_data == self._selected_data_stored
            and self._value == self._value_stored
and np.array_equal(self._drag_box, self._drag_box_stored)
```

And use it in both ifs? It will allow to remove `prev_stored`

Reply:
Yeah I was also looking at `not force`. I did some testing and it seems to give the desired effect, solving #7071 and keeping the other events.